### PR TITLE
add test to assign tag to post 

### PR DESCRIPTION
**Pull request by beblicarl**
#### Pre-Submission Checklist

-   [x] Your pull request targets the `main` branch.
-   [x] Branch name is descriptive and follows the guidelines [here](https://github.com/andela/bestpractices/wiki/Git-naming-conventions-and-best-practices).
-   [x] All tests pass . Use `git commit --amend` to amend any fixes.

#### Checklist:

-   [x] All tests are passing.
-   [x] Tested changes locally.
-   [x] Improved documentation or readme where necessary.
-   [x] Your commit messages format follows our commit message guideline [here](https://github.com/andela/bestpractices/wiki/Git-naming-conventions-and-best-practices#commits).

#### What does this PR do?

https://github.com/tbf-challenge/teamwork-be/issues/331
#### How should this be manually tested?

1. Start the test with `npm run test`
2. This should return all tests passing

#### What are the relevant issues?
https://github.com/tbf-challenge/teamwork-be/issues/331

### CODE REVIEW

**Comment by beblicarl** [comment]
> This PR is being deployed to [Railway](https://railway.app) 🚅
> 
> teamwork: [🕥 Deploying](https://railway.app/project/2c0255bb-8469-4eb0-a4d6-d4dc47f11ca2/service/b65df450-d496-4d25-b493-58cfb51e0e6b?id=7f11fe91-9d46-4aa1-b735-82f8ed23704b&environmentId=116a9949-0241-403c-8a2f-1cc61abcba47)

Is this supposed to be normal when the pr hasn't been merged?

**Comment by jocrah** [comment]
> Is this supposed to be normal when the pr hasn't been merged?

it happens when the pr is intrusive and not good enough. please delete the branch

**Comment by beblicarl** [comment]
> > Is this supposed to be normal when the pr hasn't been merged?
> 
> it happens when the pr is intrusive and not good enough. please delete the branch

🤣🤣🤣🤣🤣

**Comment by jakazzy** [comment]
@beblicarl  it seems there are changes related to the functionality. However, the PR description is only about the automated test. Please why is that so?

File: src/services/posts/assign-tag-to-post.test.js [hunk 1]
```diff
@@ -0,0 +1,117 @@
+const {expect} = require('chai')
```
Comment (jakazzy) [inline src/services/posts/assign-tag-to-post.test.js:1]: Its either linting has been disabled or we do not have rules.
Can you kindly fix the lining over here? eg `{ expect }`
Let's do same for the others

File: src/services/posts/assign-tag-to-post.test.js [hunk 2]
```diff
@@ -0,0 +1,117 @@
...
+const db = require("../../db")
+const assignTagToPost = require("./assign-tag-to-post")
+const {fixtures} = require('../../../test/utils')
+
+
+describe('ASSIGN TAG TO POST ', () => {
+	let user
+	
+	before(async ()=>{
```
Comment (jakazzy) [inline src/services/posts/assign-tag-to-post.test.js:10]: Please let's lint the entire changes. This is to ensure consistency.

**Review by jakazzy — CHANGES_REQUESTED** [review]

File: src/routes/articles/assignTagToArticle.e2e.test.js [hunk 3]
```diff
@@ -0,0 +1,140 @@
...
+
+describe('POST /articles/:id/tags', () => {
+
+	let user
+	let accessToken
+
+	before(async ()=>{
+
+		 user = await fixtures.insertUser() 
+		const body = { id: user.id, email: user.email }
```
Comment (jakazzy) [inline src/routes/articles/assignTagToArticle.e2e.test.js:13]: Hi CTO, I am unable to follow this. is there a reason why it is referred to `body`?

File: src/services/posts/assign-tag-to-post.js [hunk 4]
```diff
@@ -0,0 +1,27 @@
+const db = require("../../db")
+const {
+	TagAlreadyAssignedToPostError
+} = require("../errors")
+const customError = require("../../lib/custom-error")
+
+const uniqueErrorCode = '23505'
```
[the remark this answers is not in the document]
Comment (beblicarl) [reply src/services/posts/assign-tag-to-post.js:7]: That's the error code that is returned when you send duplicate data

File: src/routes/articles/assignTagToArticle.e2e.test.js [hunk 5]
```diff
@@ -0,0 +1,140 @@
...
+		accessToken = fixtures.generateAccessToken(
+			{user : body}
```
Comment (jakazzy) [inline src/routes/articles/assignTagToArticle.e2e.test.js:15]: Let us kindly fix the issues with lining here as well. 
Either we do not have the linter set well in the project.
For instance extra lines, and spaces, etc

File: src/routes/articles/assignTagToArticle.e2e.test.js (hunk 3 above)
Comment (beblicarl) [reply src/routes/articles/assignTagToArticle.e2e.test.js:13]: Personal choice actually. I thought of `user` being an object and `id`, `email` being fields/body of the `user` object

**Comment by jakazzy** [comment]
Since the linting seems to run through we can sort that later @beblicarl

File: src/routes/articles/assignTagToArticle.e2e.test.js (hunk 3 above)
Comment (jakazzy) [reply src/routes/articles/assignTagToArticle.e2e.test.js:13]: I think we should perhaps use intention revealing names. 
i can't think of a good one other than this name as an example 
eg. `accessTokenParams` 
if you do have one better than this i think we can use.
Any one other than body. just so we know why we are accessing those information from the user.

File: src/services/posts/assign-tag-to-post.js (hunk 4 above)
[the remark this answers is not in the document]
Comment (jakazzy) [reply src/services/posts/assign-tag-to-post.js:7]: Great!

[the remark this answers is not in the document]
Comment (jakazzy) [reply src/services/posts/assign-tag-to-post.js:7]: Thanks!

**Comment by jakazzy** [comment]
> > @beblicarl it seems there are changes related to the functionality. However, the PR description is only about the automated test. Please why is that so?
> 
> I will improve the PR description

Sure will be waiting on this @beblicarl . Kindly do let me know when it is updated.

File: src/routes/articles/assignTagToArticle.e2e.test.js (hunk 3 above)
Comment (beblicarl) [reply src/routes/articles/assignTagToArticle.e2e.test.js:13]: yes please, thanks

**Comment by beblicarl** [comment]
> > > @beblicarl it seems there are changes related to the functionality. However, the PR description is only about the automated test. Please why is that so?
> > 
> > 
> > I will improve the PR description
> 
> Sure will be waiting on this @beblicarl . Kindly do let me know when it is updated.

It has been updated please

**Review by jakazzy — APPROVED** [review]

**Pull request merged**